### PR TITLE
Make the Borgs Names Appear When they Make an Announcement

### PR DESCRIPTION
**Pull request by Cojoke-dot**
## About the PR

Makes the Communications Console display a borgs name whenever they make an announcement
## Why / Balance

Borgs using the communications console and displaying "Unknown" is inconsistent with how the console works with IDs
## Technical details

Check if the user who accessed the console has the `BorgChassisComponent`, if so then take the metadata for the borgs name and set the message author to that.
## Media

Screenshot 2024-07-16 184704

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- tweak: Borg's names are now displayed when they make an announcement on the Communications Console.

### CODE REVIEW

**Comment by Golinth** [comment]
Bless you for this, it’s been bugging me in the back of my mind for ages

File: Content.Server/Communications/CommunicationsConsoleSystem.cs [hunk 1]
```diff
@@ -255,10 +257,18 @@ private void OnSelectAlertLevelMessage(EntityUid uid, CommunicationsConsoleCompo
...
+                // User has an id
                 if (_idCardSystem.TryFindIdCard(mob, out var id))
                 {
                     author = $"{id.Comp.FullName} ({CultureInfo.CurrentCulture.TextInfo.ToTitleCase(id.Comp.JobTitle ?? string.Empty)})".Trim();
                 }
+
+                // User does not have an id and is a borg, so use the borg's name
+                if (HasComp<BorgChassisComponent>(mob))
+                {
+                    var meta = _entityManager.GetComponent<MetaDataComponent>(mob);
```
Comment (Brandon-Huu) [inline Content.Server/Communications/CommunicationsConsoleSystem.cs:269]: Why not TryComp?

Comment (Cojoke-dot) [reply Content.Server/Communications/CommunicationsConsoleSystem.cs:269]: TryComp gives a warning when getting `MetaDataComponent`
image

Comment (Brandon-Huu) [reply Content.Server/Communications/CommunicationsConsoleSystem.cs:269]: > TryComp gives a warning when getting `MetaDataComponent` image

Horrible warning. What even is a non-generic variant

Comment (deltanedas) [reply Content.Server/Communications/CommunicationsConsoleSystem.cs:269]: use Name(uid) instead

File: Content.Server/Communications/CommunicationsConsoleSystem.cs [hunk 2]
```diff
@@ -42,6 +43,7 @@ public sealed class CommunicationsConsoleSystem : EntitySystem
         [Dependency] private readonly UserInterfaceSystem _uiSystem = default!;
         [Dependency] private readonly IConfigurationManager _cfg = default!;
         [Dependency] private readonly IAdminLogManager _adminLogger = default!;
+        [Dependency] private readonly EntityManager _entityManager = default!;
```
Comment (deltanedas) [inline Content.Server/Communications/CommunicationsConsoleSystem.cs:46]: remove

**Review by deltanedas — CHANGES_REQUESTED** [review]

File: Content.Server/Communications/CommunicationsConsoleSystem.cs [hunk 3]
```diff
@@ -255,10 +256,17 @@ private void OnSelectAlertLevelMessage(EntityUid uid, CommunicationsConsoleCompo
...
                 if (_idCardSystem.TryFindIdCard(mob, out var id))
                 {
                     author = $"{id.Comp.FullName} ({CultureInfo.CurrentCulture.TextInfo.ToTitleCase(id.Comp.JobTitle ?? string.Empty)})".Trim();
                 }
+
+                // User does not have an id and is a borg, so use the borg's name
+                if (HasComp<BorgChassisComponent>(mob))
+                {
+                    author = Name(mob).Trim();
+                }
```
Comment (metalgearsloth) [inline Content.Server/Communications/CommunicationsConsoleSystem.cs:269]: Does Identity handle this?

Comment (deltanedas) [reply Content.Server/Communications/CommunicationsConsoleSystem.cs:269]: comms console uses ids not identity, it says Unknown instead of young man etc

Comment (Cojoke-dot) [reply Content.Server/Communications/CommunicationsConsoleSystem.cs:269]: borgs don't have identity

**Comment by lzk228** [comment]
it should use NameIdentifierComponent as in LogProbe cartridge, then it's not hardcoded only to borgs (#27788)

**Pull request merged**